iceoryx_posh/source/popo/ports/interface_port.cpp

```diff
@@ -19,38 +19,38 @@ namespace iox
 {
 namespace popo
 {
-InterfacePort::InterfacePort(InterfacePortData* const f_member)
-    : BasePort(f_member)
+InterfacePort::InterfacePort(InterfacePortData* const interfacePortDataPtr) noexcept
+    : BasePort(interfacePortDataPtr)
 {
 }
 
-bool InterfacePort::dispatchCaProMessage(const capro::CaproMessage& f_message)
+bool InterfacePort::dispatchCaProMessage(const capro::CaproMessage& message) noexcept
 {
-    bool returner = getMembers()->m_caproMessageFiFo.push(f_message);
+    const bool returner = getMembers()->m_caproMessageFiFo.push(message);
     if (!returner)
     {
         errorHandler(Error::kPOSH__INTERFACEPORT_CAPRO_MESSAGE_DISMISSED, nullptr, iox::ErrorLevel::SEVERE);
     }
     return returner;
 }
 
-bool InterfacePort::getCaProMessage(capro::CaproMessage& f_message)
+bool InterfacePort::getCaProMessage(capro::CaproMessage& message) noexcept
 {
-    auto msg = getMembers()->m_caproMessageFiFo.pop();
-    if (msg.has_value())
+    auto maybeMessage = getMembers()->m_caproMessageFiFo.pop();
```

Comment:
how about
```
    return getMembers()
        ->m_caproMessageFiFo.pop()
        .and_then([&message](capro::CaproMessage& receivedMessage) { message = receivedMessage; })
        .has_value();
```

Reply:
@elBoberido: Is this now free of std::function and has better or at least same performance?

Reply:
yes, it's free of std::function and the performance should also similar. I wouldn't bet the naming rights of my first born on it, though.

Reply:
You gotta prove it ;-)

Reply:
Change the signature to `cxx::optional<capro::CaproMessage> getCaProMessage()` and just write `return getMembers()->m_caproMessageFiFo.pop()`

Reply:
@elfenpiff valid point. We already switched to optional return in the other ports. But I would do that when we are harmonizing the queue interfaces. This will be one of the next tasks in #25 when we are integrating the new MPMC queue in the variant queue